12.0 ensure onchanges applied to direct report calls

gdgellatly opened the pull request:
[FIX] Ensure onchanges applied when calling report directly  …
Prior to this commit create was called in the report in order to
get the required data. However date_end was not defaulted and relied
on an onchange to get the correct date.

In order to fix we use the new Odoo tests Form functionality to
emulate calling the wizard form just as in UI.

[FIX] Add missing default in res_config_settings and bump version

gdgellatly committed:
[FIX] Aging Buckets

In certain scenarios involving mutliple payments for multiple
invoices, where the reconciliation happens together and the debit
move is the smaller item the aging duplicates its value, making
the balance greater than it should be.

This fix works by preventing moves where different lines are both partial
reconciled being grouped together by adding the lines id
to the group by clause of _show_buckets_sql_q1

gdgellatly committed:
[FIX] Ensure onchanges applied when calling report directly

Prior to this commit create was called in the report in order to
get the required data. However date_end was not defaulted and relied
on an onchange to get the correct date.

In order to fix we use the new Odoo tests Form functionality to
emulate calling the wizard form just as in UI.

gdgellatly committed:
[FIX] Add missing default and bump version

MiquelRForgeFlow commented on partner_statement/report/activity_statement.py:
```diff
@@ -140,9 +141,9 @@ def _get_report_values(self, docids, data):
             data = {}
         if 'company_id' not in data:
-            wiz = self.env["activity.statement.wizard"].with_context(
+            wiz = Form(self.env["activity.statement.wizard"].with_context(
```
Don't! This `Form` server was created only for testing. That's why it is found in odoo.tests 😏

pedrobaeza replied:
@mreficent it's on tests, but it's used in other parts in Odoo (see vendor bill import). I see this better than simply call all the onchanges you know (and leaving unknown onchanges), or to depend on `onchange_helper` module, that is buggy in some cases (when there's a context in fields).

gdgellatly replied:
@mreficent In this particular case, it is the precise behaviour desired. As though the user opened the form, the defaults were populated, and triggered the onchanges. I figured if it is considered good enough for core for basically the same cases, then it must be OK.

jjscarafia committed:
[12.0][ADD] mis_builder_cash_flow module

etobella committed:
[IMP] mis_builder_cash_flow: Add tests

gdgellatly committed:
Merge branch '12.0' of https://github.com/OCA/account-financial-reporting into OCA-12.0

gdgellatly committed:
Merge branch 'OCA-12.0' into 12.0-ensure-onchanges-applied

Commit:
Translated using Weblate (Italian)

Currently translated at 86.1% (68 of 79 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-partner_statement
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-partner_statement/it/

Commit:
Translated using Weblate (Italian)

Currently translated at 87.3% (69 of 79 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-partner_statement
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-partner_statement/it/

Commit:
Translated using Weblate (Italian)

Currently translated at 87.3% (69 of 79 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-partner_statement
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-partner_statement/it/

Commit:
Translated using Weblate (Italian)

Currently translated at 87.3% (69 of 79 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-partner_statement
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-partner_statement/it/

oca-travis committed:
[UPD] Update mis_builder_cash_flow.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

gdgellatly commented:
Wow I must have messed this up sometime, guess we just forked and kept going seperately.
